compiler/packages/babel-plugin-react-compiler/src/__tests__/fixtures/compiler/array-map-mutable-array-non-mutating-lambda-mutated-result.expect.md

```diff
@@ -29,8 +29,9 @@ function Component(props) {
   if ($[0] === Symbol.for("react.memo_cache_sentinel")) {
     const x = [{}];
     const y = x.map((item) => item);
-    y[0].flag = true;
+
     t0 = [x, y];
+    y[0].flag = true;
```

Comment:
This happens because we allow LoadLocal to be reordered relative to other instructions that use the loaded variable. I think we can make it safe to do LoadLocal reordering (by moving until after InferMutableLifetimes and sequencing all mutations of a given scope id), but it's definitely safer to not reorder them.